src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -195,12 +220,31 @@ async def handle_bsp_scalac_options_request(
     )
     coarsened_targets = await Get(CoarsenedTargets, Addresses(tgt.address for tgt in targets))
     resolve = await Get(CoursierResolveKey, CoarsenedTargets, coarsened_targets)
+    lockfile = await Get(CoursierResolvedLockfile, CoursierResolveKey, resolve)
+
+    resolve_digest = await Get(
+        Digest,
+        CreateDigest([FileEntry(entry.file_name, entry.file_digest) for entry in lockfile.entries]),
+    )
+
+    resolve_digest = await Get(
+        Digest, AddPrefix(resolve_digest, f"jvm/resolves/{resolve.name}/lib")
+    )
+
+    workspace.write_digest(resolve_digest, path_prefix=".pants.d/bsp")
+
+    classpath = [
+        build_root.pathlib_path.joinpath(
+            f".pants.d/bsp/jvm/resolves/{resolve.name}/lib/{entry.file_name}"
```

Comment:
Would be good to dedupe with the other string calculation mentioned above.

Reply:
Agreed. One of the follow-ons is going to merge all of this path-related code to some extent.